### PR TITLE
Remove float support

### LINKED ISSUE #31: We need to reconsider supporting FLOAT types in our SQL spec
We should decide on how to deal with the discovered FLOAT problem about different CPU architectures doing rounding in different ways.

There's an ongoing discussion [in Notion](https://www.notion.so/textile/Floating-point-math-is-hard-7c3e39f4062d4698a29584bb24655b8e); we should come up with a conclusion about:
- What we'll do in the SQL spec about this.
- If we are dropping support for FLOATs, then that would mean that existing history will be affected (e.g.: existing tables with FLOAT columns will become invalid, maybe there aren't that many, but it would be a side-effect to be aware of).

Spec issue: https://github.com/tablelandnetwork/docs/issues/27

cc @sanderpick @carsonfarmer @brunocalza @dtbuchholz @andrewxhill @asutula @joewagner @awmuncy

### PR DESCRIPTION
**Pull request by brunocalza**
# Summary

Removes support for floating point numbers because they have non deterministic behavior

Closes #31 

# Context

See [Spec: we need to solve non-determinism for floating point columns](https://github.com/tablelandnetwork/docs/issues/27)

# Implementation overview

- Remove the `REAL` type for `CREATE` statements
- Remove the `REAL` and `NUMERIC` type for `CAST` expressions
- Adds an custom error when a `FLOAT` token is seen

# Implementation details and review orientation

na

### CODE REVIEW

File: grammar.y [hunk 1]
```diff
@@ -104,7 +104,7 @@ func isRowID(column Identifier) bool {
 %token ERROR 
 %token <empty> TRUE FALSE NULL AND
 %token <empty> '(' ',' ')' '.' ';'
-%token <empty> NONE INTEGER NUMERIC REAL TEXT CAST AS
```
Comment (brunocalza) [inline grammar.y:107]: Remove `NUMERIC` and `REAL` tokens

File: grammar.y [hunk 2]
```diff
@@ -829,9 +829,7 @@ between_op:
 convert_type:
```
Comment (brunocalza) [inline grammar.y:829]: Remove `NUMERIC` and `REAL` from `CAST` expression

File: grammar.y [hunk 3]
```diff
@@ -1046,7 +1044,6 @@ column_def:
 type_name:
```
Comment (brunocalza) [inline grammar.y:1044]: Remove `REAL` from `CREATE`

File: lexer.go [hunk 4]
```diff
@@ -15,8 +15,6 @@ var keywords = map[string]int{
 	"NULL":       NULL,
 	"NONE":       NONE,
 	"INTEGER":    INTEGER,
-	"NUMERIC":    NUMERIC,
```
Comment (brunocalza) [inline lexer.go:18]: Remove keywords from the lexers. That means that the lexer won't emit tokens for those keywords, which will probably result in a syntax error

File: grammar.y [hunk 5]
```diff
@@ -1165,6 +1162,7 @@ numeric_literal:
   }
 | FLOAT
   {
+    yylex.(*Lexer).AddError(&ErrNumericLiteralFloat{Value: $1})
```
Comment (brunocalza) [inline grammar.y:1165]: We are still parsing literal float numbers but throwing an error now. Another option would be to not be able to understand float numbers and throw a syntax error. But that would imply a bigger and risker change in the lexer. I think this is fine. Maybe we will still need to parse floats for some reason, better to keep that logic.

Comment (jsign) [reply grammar.y:1165]: Agree.

**Review by jsign — APPROVED** [review]

**Pull request merged**